### PR TITLE
test: fix local flakiness while keeping unique identifiers

### PR DESCRIPTION
**Pull request by nopcoder**
Fix https://github.com/treeverse/lakeFS/issues/4736
Fix https://github.com/treeverse/lakeFS/issues/5100

Also updated mem block adapter to verify the storage namespace - align the behaviour as other block adapters.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Sorry... this will take me a while to review.

AFAICT there are 2 refactorings here added on top of the fix, which is what makes it hundreds of lines.  If you could split them off into separate commits, I might be able to get through more quickly.  I'll still do it of course, just a matter of how long it'll take me!

File: pkg/api/controller_test.go [hunk 1]
```diff
@@ -238,10 +238,11 @@ func TestController_LogCommitsMissingBranch(t *testing.T) {
 	clt, deps := setupClientWithAdmin(t)
 	ctx := context.Background()
 
-	_, err := deps.catalog.CreateRepository(ctx, "repo1", onBlock(deps, "ns1"), "main")
+	repo := testUniqueRepoName()
```
Comment (arielshaqed) [inline pkg/api/controller_test.go:241]: This change (throughout) is _gold_ 🥇 !

**Review by arielshaqed — APPROVED** [review]
THanks!

**Comment by nopcoder** [comment]
> Sorry... this will take me a while to review.
> 
> AFAICT there are 2 refactorings here added on top of the fix, which is what makes it hundreds of lines. If you could split them off into separate commits, I might be able to get through more quickly. I'll still do it of course, just a matter of how long it'll take me!

Thanks @arielshaqed I can split the mem adapter change from the unique repository change.

**Comment by arielshaqed** [comment]
It's good :-)  THANKS!

**Pull request merged**